packages/expo-audio/android/src/main/java/expo/modules/audio/service/AudioControlsService.kt

```diff
@@ -0,0 +1,212 @@
+package expo.modules.audio.service
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.os.Binder
+import android.os.Build
+import android.os.IBinder
+import androidx.annotation.OptIn
+import androidx.core.app.NotificationCompat
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.session.MediaSession
+import androidx.media3.session.MediaSessionService
+import androidx.media3.session.MediaStyleNotificationHelper
+import expo.modules.audio.AudioPlayer
+import expo.modules.audio.Metadata
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import java.net.URL
+
+@OptIn(UnstableApi::class)
+class AudioControlsService : MediaSessionService() {
+  private val binder = AudioControlsBinder()
+  private var mediaSession: MediaSession? = null
+  private var currentMetadata: Metadata? = null
+  private var currentPlayer: AudioPlayer? = null
+  private val scope = CoroutineScope(Dispatchers.IO)
+  private var currentArtworkUrl: URL? = null
+  private var currentArtwork: Bitmap? = null
```

Comment:
Consider encapsulating all the var's into a single data class and modifying it by calling `copy` with new values. I think it would be more readable.